### PR TITLE
Display app conditions timestamp in CLI and UI. (#737)

### PR DESCRIPTION
**Pull request by dgoodwin**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Optional. My organization is added to the README.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

Followup to #2417 exposing the new timestamp in UI and CLI. Not sure if formatting is desired for this, let me know, right now they're just raw timestamps.

There appears to be a problem with the conditions in the UI, they are not visible by default, you need to navigate to App Details and then append conditions=true to the URL to see them. I'm not sure how to fix this yet, I don't know anything about React, but may take a look in future.

### CODE REVIEW

**Comment by alexec** [comment]
Thank you @dgoodwin - you'll need to fix the UI issues before we can merge please.

File: ui/src/app/applications/components/application-conditions/application-conditions.tsx [hunk 1]
```diff
@@ -15,7 +15,8 @@ export const ApplicationConditions = ({conditions}: {conditions: models.Applicat
                         <div className={`argo-table-list__row application-conditions__condition application-conditions__condition--${getConditionCategory(condition)}`} key={index}>
                             <div className='row'>
                                 <div className='columns small-2'>{condition.type}</div>
-                                <div className='columns small-10'>{condition.message}</div>
+                                <div className='columns small-8'>{condition.message}</div>
+                                <div className='columns small-2'>{condition.lastTransitionTime}</div>
```
Comment (alexmt) [inline ui/src/app/applications/components/application-conditions/application-conditions.tsx]: Please use `<Timestamp date={condition.lastTransitionTime}/>` to render date in the user time zone. Example: https://github.com/argoproj/argo-cd/blob/a0f39034182e4a82f53bb5ce684e4e0d45bcb1c8/ui/src/app/applications/components/application-deployment-history/application-deployment-history.tsx#L37

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you for contribution, @dgoodwin ! Added one minor comment.

**Comment by alexmt** [comment]
Just noticed that CI failed:  https://circleci.com/gh/argoproj/argo-cd/7697?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```
 TS2339: Property 'lastTransitionTime' does not exist on type 'ApplicationCondition'.
```

To fix it add `lastTransitionTime: string`  to the `ApplicationCondition` definition in https://github.com/argoproj/argo-cd/blob/73590e1a3982501865577c2a2c9ec8794391b813/ui/src/app/shared/models.ts#L326

**Comment by dgoodwin** [comment]
Thanks for taking the time to point me at these things, I am completely new to react and frontend js. Will get these fixed and updated asap.

**Comment by alexec** [comment]
CI is failing because UI is incorrectly formatted. We use Prettier to format all UI code in an opinionated fashion. Just `cd ui` and run `yarn lint --fix`, commit and push the changes.

**Comment by dgoodwin** [comment]
Updated!

**Review by alexmt — APPROVED** [review]
Thank you! LGTM

**Pull request merged**